common/rpc/interceptor/caller_info.go

```diff
@@ -35,32 +35,48 @@ import (
 )
 
 type (
-	CallerInfoInterceptor struct {
+	CallerInfo struct {
 		namespaceRegistry namespace.Registry
 	}
 )
 
-var _ grpc.UnaryServerInterceptor = (*CallerInfoInterceptor)(nil).Intercept
+var _ grpc.UnaryServerInterceptor = (*CallerInfo)(nil).Intercept
 
-func NewCallerInfoInterceptor(
+func NewCallerInfo(
 	namespaceRegistry namespace.Registry,
-) *CallerInfoInterceptor {
-	return &CallerInfoInterceptor{
+) *CallerInfo {
+	return &CallerInfo{
 		namespaceRegistry: namespaceRegistry,
 	}
 }
 
-func (i *CallerInfoInterceptor) Intercept(
+func (i *CallerInfo) Intercept(
 	ctx context.Context,
 	req interface{},
 	info *grpc.UnaryServerInfo,
 	handler grpc.UnaryHandler,
 ) (interface{}, error) {
+	ctx = PopulateCallerInfo(
+		ctx,
+		func() string { return string(MustGetNamespaceName(i.namespaceRegistry, req)) },
+		func() string { return api.MethodName(info.FullMethod) },
```

Comment:
Why do these need to be functions? You could just pass the strings in.

Reply:
Mainly because if the namespace name is already set (the `CallerName` header) then I want to avoid hitting the namespace cache again as that may be expensive if it reads-through. It's probably not a very necessary optimization, but I wanted to preserve the behavior of the original code, which only does the lookup if the header is unset. Similar rationale for the method name; I don't think `strings.LastIndex` is particularly expensive but no reason to look it up if the header is already set.

Reply:
I see, makes sense to me.